source/transactions-convenient-api/transactions-convenient-api.rst

```diff
@@ -268,7 +268,14 @@ This method can be expressed by the following pseudo-code:
                      * being retried (see: DRIVERS-601) */
                     this.commitTransaction();
                 } catch (error) {
-                    if (error.hasErrorLabel("UnknownTransactionCommitResult") &&
+                    /* Note: a maxTimeMS error will have the MaxTimeMSExpired
+                     * code (50) and can be reported as a top-level error or
+                     * inside writeConcernError, ie:
+                     * {ok:0, code: 50, codeName: "MaxTimeMSExpired"}
+                     * {ok:1, writeConcernError: {code: 50, codeName: "MaxTimeMSExpired"}}
```

Comment:
This may be off topic, but is there a particular reason we've resorted to checking error codes instead of relying on labels? Was there something about maxTimeMS that necessitated using an error code?

Reply:
The server does not return the UnknownTransactionCommitResult error label. I can open a ticket for them to add it in future versions though.

Reply:
If they're not able to make the change for 4.2, I suppose it doesn't much matter since we'll need to add code to check the error code directly. I think this is worth raising with the server team, though, as I was under the impression this was the intended purpose for error labels (getting away from requiring drivers to inspect error codes).

Reply:
@ShaneHarvey: Did you ever end up creating a SERVER ticket for this? If so, can you cross-reference it with [SPEC-1206](https://jira.mongodb.org/browse/SPEC-1206)?

Reply:
Thanks the reminder! I just opened [SERVER-41125](https://jira.mongodb.org/browse/SERVER-41125) for this and linked it to SPEC-1206.